plugins/database/src/db_plugin.cpp

```diff
@@ -13912,28 +13913,49 @@ namespace
 
         // Modify ticket operation.
         if (std::strcmp(_op_name, "mod") == 0) {
+            int i{0};
+            char myTime[TIME_LEN]{}; // NOLINT(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+
             if (std::strcmp(_arg3, "uses") == 0) {
-                cllBindVars[0] = _arg4;
-                cllBindVars[1] = ticket_id_string.c_str();
-                cllBindVarCount = 2;
+                getNowStr(myTime); // NOLINT(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+                i = 0;
+                // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+                cllBindVars[i++] = _arg4;
+                // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+                cllBindVars[i++] = myTime; // NOLINT(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+                // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-constant-array-index)
+                cllBindVars[i++] = ticket_id_string.c_str();
+                cllBindVarCount = i;
 
-                return execute_sql("update R_TICKET_MAIN set uses_limit = ? where ticket_id = ?");
+                return execute_sql("update R_TICKET_MAIN set uses_limit = ?, modify_ts = ? where ticket_id = ?");
```

Comment:
I am not sure. @korydraughn do you have an answer?

Reply:
It looks like we do need to call `_rollback()`, but let's confirm that first. It's possible the rollback is handled earlier in the stack _(I don't think it is though)_.

@Nishant2022 Review the logic of `db_mod_ticket_op()` in db_plugin.cpp and let us know what you find.

Reply:
There does not appear to be any rollback logic in `db_mod_ticket_op()`.

We only need to roll back in the instances when two SQL statements are required correct? Such as when we insert and then update the modify time. If the update modify time fails, then we roll back the changes made during the insert.

However, I do not believe we need to do it when there is only one SQL statement because if it fails, we return the error code and nothing changes, correct?

Reply:
I'd argue it is more about the auto-commit option on every database connection.

Disabling auto-commit is the default for database connections in iRODS AFAIK. For that reason, we need to make sure we keep the database happy by properly committing or rolling back at the appropriate time.

This also helps the next developer who encounters the code.